src/providers/execution/contracts/lido_locator.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from functools import lru_cache
+
+from eth_typing import ChecksumAddress
+from web3.types import BlockIdentifier
+
+from src.providers.execution.base_interface import ContractInterface
+
+
+logger = logging.getLogger(__name__)
+
+
+class LidoLocatorContract(ContractInterface):
+    abi_path = './assets/LidoLocator.json'
+
+    @lru_cache(maxsize=1)
+    def lido(self, block_identifier: BlockIdentifier = 'latest') -> ChecksumAddress:
```

Comment:
Wait, the cache here for the "block_identifier = 'latest'" will break the thing?

Reply:
Yeah, possibly it can break something.
Fixed